AUTH /api/clusters returns response with panic error

Call to https://auth.prod-preview.openshift.io/api/clusters

Returns response with below invalid json format:
```
{
  "data": []
}
{
  "errors": []
}
```

Where errors contains panic error stack trace as below:
```
panic: net/http: abort Handler
panic(0xbc30c0, 0xc000030ab0)
	/usr/lib/golang/src/runtime/panic.go:513 +0x1b9
net/http/httputil.(*ReverseProxy).ServeHTTP(0xc0005c9a90, 0xdff980, 0xc0008723c0, 0xc0001c3400)
	/usr/lib/golang/src/net/http/httputil/reverseproxy.go:284 +0x1058
github.com/fabric8-services/fabric8-auth/vendor/github.com/fabric8-services/fabric8-common/httpsupport.route(0xe00ec0, 0xc0003af0a0, 0xcce642, 0x10, 0x0, 0x0, 0x0, 0x0, 0xcce642, 0x10)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/vendor/github.com/fabric8-services/fabric8-common/httpsupport/proxy.go:85 +0x2d0
github.com/fabric8-services/fabric8-auth/vendor/github.com/fabric8-services/fabric8-common/httpsupport.RouteHTTP(0xe00ec0, 0xc0003af0a0, 0xcce642, 0x10, 0x0, 0x0, 0x0, 0xc0003af0a0, 0xc9a240)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/vendor/github.com/fabric8-services/fabric8-common/httpsupport/proxy.go:43 +0x7d
github.com/fabric8-services/fabric8-auth/controller.(*ClustersController).Show(0xc0003d0680, 0xc0003af0a0, 0xc0001c3400, 0xc0000863c0)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/controller/clusters.go:33 +0x8e
github.com/fabric8-services/fabric8-auth/app.MountClustersController.func1(0xe00300, 0xc0008f2480, 0xdfdf40, 0xc00037db80, 0xc0001c3400, 0x0, 0x0)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/app/controllers.go:132 +0xdd
github.com/fabric8-services/fabric8-auth/vendor/github.com/goadesign/goa/middleware/security/jwt.New.func1.1(0xe00300, 0xc0008f2240, 0xdfdf40, 0xc00037db80, 0xc0001c3400, 0xe00300, 0xc0008f2240)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/vendor/github.com/goadesign/goa/middleware/security/jwt/jwt.go:123 +0x4c3
github.com/fabric8-services/fabric8-auth/app.handleSecurity.func1(0xe00300, 0xc0008f2240, 0xdfdf40, 0xc00037db80, 0xc0001c3400, 0xc222a0, 0xc00037db80)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/app/security.go:49 +0x177
github.com/fabric8-services/fabric8-auth/app.handleClustersOrigin.func1(0xe00300, 0xc0008f2090, 0xdfdf40, 0xc00037db80, 0xc0001c3400, 0x0, 0x0)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/app/controllers.go:148 +0x4e4
github.com/fabric8-services/fabric8-auth/vendor/github.com/goadesign/goa.(*Controller).MuxHandler.func1.1(0xe00300, 0xc0008f2090, 0xdfdf40, 0xc00037db80, 0xc0001c3400, 0x0, 0x0)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/vendor/github.com/goadesign/goa/service.go:270 +0x9b
github.com/fabric8-services/fabric8-auth/log.LogRequest.func1.1(0xe00300, 0xc0008f2090, 0xdfdf40, 0xc00037db80, 0xc0001c3400, 0xc0008f2000, 0xe00300)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/log/log_request.go:90 +0xd0a
github.com/fabric8-services/fabric8-auth/authorization/token/manager.InjectTokenManager.func1.1(0xe00300, 0xc000939fb0, 0xdfdf40, 0xc00037db80, 0xc0001c3400, 0xc0005c9680, 0x0)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/authorization/token/manager/token_manager.go:170 +0x98
github.com/fabric8-services/fabric8-auth/goamiddleware.handler.func1(0xe00300, 0xc000939d40, 0xdfdf40, 0xc00037db80, 0xc0001c3400, 0xd9, 0x0)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/goamiddleware/jwt_token_context.go:49 +0x272
github.com/fabric8-services/fabric8-auth/vendor/github.com/goadesign/goa/middleware.Recover.func1.1(0xe00300, 0xc000939d40, 0xdfdf40, 0xc00037db80, 0xc0001c3400, 0x0, 0x0)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/vendor/github.com/goadesign/goa/middleware/recover.go:37 +0x93
github.com/fabric8-services/fabric8-auth/jsonapi.ErrorHandler.func1.1(0xe00300, 0xc000939d40, 0xdfdf40, 0xc00037db80, 0xc0001c3400, 0x1, 0xc00048fb40)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/jsonapi/error_handler.go:37 +0xb4
github.com/fabric8-services/fabric8-auth/vendor/github.com/goadesign/goa/middleware/gzip.Middleware.func2.1(0xe00300, 0xc000939d40, 0xdfdf40, 0xc00037db80, 0xc0001c3400, 0x0, 0x0)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/vendor/github.com/goadesign/goa/middleware/gzip/middleware.go:98 +0x3a3
github.com/fabric8-services/fabric8-auth/log.LogRequest.func1.1(0xe00300, 0xc000939d40, 0xdfdf40, 0xc00037db80, 0xc0001c3400, 0xc00046f830, 0xe00300)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/log/log_request.go:90 +0xd0a
github.com/fabric8-services/fabric8-auth/vendor/github.com/goadesign/goa/middleware.RequestIDWithHeaderAndLengthLimit.func1.1(0xe00300, 0xc000939c80, 0xdfdf40, 0xc00037db80, 0xc0001c3400, 0xc000939bf0, 0xe00300)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/vendor/github.com/goadesign/goa/middleware/request_id.go:63 +0x144
github.com/fabric8-services/fabric8-auth/vendor/github.com/goadesign/goa.(*Controller).MuxHandler.func1(0xdff880, 0xc0005e37a0, 0xc0001c3400, 0xc000939bf0)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/vendor/github.com/goadesign/goa/service.go:303 +0x31e
github.com/fabric8-services/fabric8-auth/vendor/github.com/goadesign/goa.(*mux).Handle.func1(0xdff880, 0xc0005e37a0, 0xc0001c3400, 0x0)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/vendor/github.com/goadesign/goa/mux.go:59 +0x1c4
github.com/fabric8-services/fabric8-auth/vendor/github.com/dimfeld/httptreemux.(*TreeMux).ServeHTTP(0xc0003b4640, 0xdff880, 0xc0005e37a0, 0xc0001c3400)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/vendor/github.com/dimfeld/httptreemux/router.go:200 +0xe5
github.com/fabric8-services/fabric8-auth/vendor/github.com/goadesign/goa.(*mux).ServeHTTP(0xc0003ea260, 0xdff880, 0xc0005e37a0, 0xc0001c3400)
	/tmp/go/src/github.com/fabric8-services/fabric8-auth/vendor/github.com/goadesign/goa/mux.go:85 +0x4c
net/http.(*ServeMux).ServeHTTP(0x1401ee0, 0xdff880, 0xc0005e37a0, 0xc0001c3400)
	/usr/lib/golang/src/net/http/server.go:2361 +0x127
net/http.serverHandler.ServeHTTP(0xc00009d5f0, 0xdff880, 0xc0005e37a0, 0xc0001c3400)
	/usr/lib/golang/src/net/http/server.go:2741 +0xab
net/http.(*conn).serve(0xc000872000, 0xe00240, 0xc00037dac0)
	/usr/lib/golang/src/net/http/server.go:1847 +0x646
created by net/http.(*Server).Serve
	/usr/lib/golang/src/net/http/server.go:2851 +0x2f5
```

--- a/httpsupport/proxy.go
+++ b/httpsupport/proxy.go
@@ -1,10 +1,7 @@
 package httpsupport
 
 import (
-	"bytes"
-	"compress/gzip"
 	"context"
-	"io/ioutil"
 	"net/http"
 	"net/http/httputil"
 	"net/url"
@@ -80,12 +77,7 @@ func route(ctx context.Context, targetHost string, targetPath *string, options .
 		opt(proxy)
 	}
 
-	if strings.Contains(req.Header.Get("Accept-Encoding"), "gzip") {
-		gzr := gunzipResponseWriter{ctx: ctx, ResponseWriter: rw, targetURL: *req.URL}
-		proxy.ServeHTTP(gzr, req.Request)
-	} else {
-		proxy.ServeHTTP(rw, req.Request)
-	}
+	proxy.ServeHTTP(rw, req.Request)
 
 	return nil
 }
@@ -140,45 +132,6 @@ func newDirector(ctx context.Context, originalRequestData *goa.RequestData, targ
 			"target_string":    target.String(),
 		}, "Routing %s to %s", originalReqString, targetReqString)
 	}
-}
-
-type gunzipResponseWriter struct {
-	http.ResponseWriter
-	ctx       context.Context
-	targetURL url.URL
-}
-
-func (w gunzipResponseWriter) Write(b []byte) (int, error) {
-	// Write gunzipped data to the client
-	gr, err := gzip.NewReader(bytes.NewBuffer(b))
-	if err != nil {
-		return 0, err
-	}
-	defer func() {
-		err := gr.Close()
-		if err != nil {
-			log.Error(w.ctx, map[string]interface{}{
-				"err":        err,
-				"target_url": w.targetURL.String(),
-			}, "unable to close gzip writer while serving request in proxy")
-		}
-	}()
-	data, err := ioutil.ReadAll(gr)
-	if err != nil {
-		return 0, err
-	}
-	return w.ResponseWriter.Write(data)
-}
-
-func (w gunzipResponseWriter) WriteHeader(code int) {
-	w.Header().Del("Content-Length")
-	// Remove duplicated headers
-	for key, value := range w.Header() {
-		if len(value) > 0 {
-			w.Header().Set(key, value[0])
-		}
-	}
-	w.ResponseWriter.WriteHeader(code)
 }
 
 func singleJoiningSlash(a, b string) string {

--- a/httpsupport/proxy_blackbox_test.go
+++ b/httpsupport/proxy_blackbox_test.go
@@ -1,10 +1,12 @@
 package httpsupport_test
 
 import (
+	"bytes"
 	"compress/gzip"
 	"context"
 	"fmt"
 	"io"
+	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
 	"net/url"
@@ -55,6 +57,8 @@ func TestProxy(t *testing.T) {
 	rw = httptest.NewRecorder()
 	req, err = http.NewRequest("POST", u.String(), nil)
 	require.NoError(t, err)
+	// Do not suppress panics
+	req = req.WithContext(context.WithValue(context.Background(), http.ServerContextKey, "ProxyTest"))
 
 	ctx = context.Background()
 	goaCtx = goa.NewContext(goa.WithAction(ctx, "ProxyTest"), rw, req, url.Values{})
@@ -68,7 +72,19 @@ func TestProxy(t *testing.T) {
 	assert.Equal(t, "proxyTest", rw.Header().Get("Custom-Test-Header"))
 	body, err = httpsupport.ReadBody(rw.Result().Body)
 	require.NoError(t, err)
-	assert.Equal(t, veryLongBody, body)
+	assert.Equal(t, veryLongBody, unzip(t, body))
+}
+
+func unzip(t *testing.T, s string) string {
+	gr, err := gzip.NewReader(bytes.NewBuffer([]byte(s)))
+	require.NoError(t, err)
+	defer func() {
+		err := gr.Close()
+		require.NoError(t, err)
+	}()
+	data, err := ioutil.ReadAll(gr)
+	require.NoError(t, err)
+	return string(data)
 }
 
 func TestProxyWithOptions(t *testing.T) {

